etc/leapp/actor_conf.d/rhui.yml

```diff
@@ -0,0 +1,12 @@
+rhui:
+  use_config : True
+  source_clients: [ rh-amazon-rhui-client ]
+  target_clients: [ rh-amazon-rhui-client ]
+  cloud_provider: aws
+  upgrade_files:
```

Comment:
This example is looking good! I also know of some use cases that need to specify the CA certificate used for the dnf calls to the RHUI repo, but perhaps that configuration is encapsulated in the leapp-aws.repo file...?

Reply:
Do you mean the `sslcacert`, `sslclientcert` and `sslclientkey` fields in repository definition? If yes, then yes, this is doable with the proposed configuration. You just specify all the `ssl*` fields in a repofile provided in `upgrade_files` as you would normally do. Then you need to list the certificates/keys also in `upgrade_files`, together with an information on where to place them. For example, your repofile `/root/my-upgrade-repos.repo` would contain a repository with:
```
sslclientcert=/etc/pki/rhui/product/content-rhel8.crt
```
Then you need tell leapp that it should grab the `content-rhel8.crt` somewhere from the source system and place it at `/etc/pki/rhui/product/content-rhel8.crt`.

The configuration would then look like:
```yaml
rhui:
  use_config : True
  source_clients: [  rhel8-client ]   # what client to remove from the source system
  target_clients: [ rhel9-client ]  # what client should be installed during the upgrade
  cloud_provider: aws  # for leapp to use internally 
  upgrade_files:
    /root/my-upgrade-repos.repo : /etc/yum.repos.d  # during upgrade, place this repofile into /etc/yum.repos.d
    /root/content-rhel8.crt : /etc/pki/rhui/product/  # during the upgrade, place this key here. Same path as in sslclientcert in the repo
   #  .... other keys/certs mentioned in the repofile, or otherwise important for leapp to access target OS content
```
This should enable basically anyone running a RHUI system to upgrade, assuming that provider's rhui clients do not use some eccentric python dependencies (although that should be fine too, I just haven't tested these setups).

You get all the necessary `upgrade_files` by spawning a RHEL VM running the target OS, and copying the files, keys and certs provided by the RHUI client. There are also other ways, but this is likely the simplest one. If you are upgrading an entire fleet, you naturally need to spawn the target machine only once.

Also note that if you rename the repositories in the repofile for some reason, and then use these new names in `enabled_target_repositories`, the DNF transaction check leapp does might fail. Therefore, use the same repoids as the target client uses and everything should be fine. If you would like to test this, please ping me with any questions. It is always nice to get feedback before we merge this :grinning:.